From fee3d6501992e40a91bb2d237af10923485cb826 Mon Sep 17 00:00:00 2001
From: Rajat Soni <quic_rajson@quicinc.com>
Date: Wed, 17 Aug 2022 10:51:12 +0530
Subject: [PATCH] ath12k: Fix memory leak

During rmmod, insmod and in SSR case memory leak is observed.
Freed buff->payload in ath12k_dbring_buf_cleanup().
In ath12k_dp_cc_cleanup(), return is there after variable declaration
that don't allow func to do cleanup and memory is not getting freed.
So removed return.

Signed-off-by: Rajat Soni <quic_rajson@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dbring.c | 1 +
 drivers/net/wireless/ath/ath12k/dp.c     | 2 --
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/dbring.c b/drivers/net/wireless/ath/ath12k/dbring.c
index 026eed9..1c7855e 100644
--- a/drivers/net/wireless/ath/ath12k/dbring.c
+++ b/drivers/net/wireless/ath/ath12k/dbring.c
@@ -349,6 +349,7 @@ void ath12k_dbring_buf_cleanup(struct ath12k *ar, struct ath12k_dbring *ring)
 		idr_remove(&ring->bufs_idr, buf_id);
 		dma_unmap_single(ar->ab->dev, buff->paddr,
 				 ring->buf_sz, DMA_FROM_DEVICE);
+		kfree(buff->payload);
 		kfree(buff);
 	}
 
diff --git a/drivers/net/wireless/ath/ath12k/dp.c b/drivers/net/wireless/ath/ath12k/dp.c
index 6166ea4..32a2f8f 100644
--- a/drivers/net/wireless/ath/ath12k/dp.c
+++ b/drivers/net/wireless/ath/ath12k/dp.c
@@ -1124,8 +1124,6 @@ static void ath12k_dp_cc_cleanup(struct ath12k_base *ab)
 	struct sk_buff *skb;
 	int i;
 
-	return;
-
 	if (!dp->spt_info)
 		return;
 
-- 
2.17.1

